internal/firstoutboundmiddleware/middleware.go

```diff
@@ -92,10 +85,4 @@ func updateStream(ctx context.Context, req *transport.StreamRequest, out transpo
 	if namer, ok := out.(transport.Namer); ok {
 		req.Meta.Transport = namer.TransportName()
 	}
-
-	// Update the caller procedure to the current procedure making this request
-	call := encoding.CallFromContext(ctx)
-	if call != nil {
-		req.Meta.CallerProcedure = call.Procedure()
-	}
```

Comment:
I would recommend to comment the code and not to delete it:
- You can comment it and add a TODO (with issue number) on why it has been commented
- The benefits of commented the code is if someone starts to look to the `CallerProcedure` fields, he will see comments/indications that the feature is still WIP. By removing it, you don't give this indication to the reader and people might thing something is missing (especially since not everything from this feature has been removed).

Reply:
Thank your the suggestion @Dogild, I usually don't prefer keeping the commented code in production. We can always  merge this change back once we are ready with the fix, in this case, I am not sure when I am going to merge this change back.